src/nm.c

```diff
@@ -530,6 +530,17 @@ write_nm_conf_access_point(NetplanNetDefinition* def, const char* rootdir, const
         g_string_append_printf(s, "uuid=%s\n", uuidstr);
     }
 
+    if (def->activation_mode) {
+        /* XXX: For now NetworkManager only supports the "manual" activation
+         * mode */
+        if (g_strcmp0(def->activation_mode, "off") == 0) {
+            g_fprintf(stderr, "ERROR: %s: NetworkManager definitions do not support activation_mode off\n", def->id);
```

Comment:
I'd rather like to check for the supported feature (i.e. whitelist) here, and fail for all others, instead of the other way around, to avoid unexpected behavior once new values are added to "activation-mode". Also I'd mention the setting name (activation-mode) vs variable name (activation_mode) in the error message.
```suggestion
        if (!!g_strcmp0(def->activation_mode, "manual")) {
            g_fprintf(stderr, "ERROR: %s: NetworkManager definitions do not support activation-mode %s\n", def->id, def->activation_mode);
```